### PR TITLE
Added setCurrentEvent to the group gateway and createEventTracker to the event service

### PR DESCRIPTION
**Pull request by youssef-attia**
### Summary 

This pull request is the first step towards implementing feature Foo

- [x] implemented setCurrentEvent and createEventTracker

### Blockers 

- A newly discovered dependency that hasn’t been addressed

### Breaking Changes  

- Database schema change (anything that changes TypeORM entities)
- Other change that could cause problems (Detailed in notes)

### CODE REVIEW

**Review by cathli66 — APPROVED** [review]
Looks great! Maybe there is some way to reduce code reuse in the try catch block?

**Review by neketka — CHANGES_REQUESTED** [review]
Please resolve the CI issues before merging.

File: server/src/group/group.gateway.ts [hunk 1]
```diff
@@ -24,6 +25,7 @@ export class GroupGateway {
   constructor(
     private clientService: ClientService,
     private groupService: GroupService,
+    private eventService: EventService,
```
Comment (neketka) [inline server/src/group/group.gateway.ts:28]: You added EventService here, but GroupModule does not import EventModule in its 'imports' array. This code will not start if you were to run 'docker compose up'.

**Review by neketka — CHANGES_REQUESTED** [review]

File: server/src/group/group.gateway.ts [hunk 2]
```diff
@@ -76,5 +76,32 @@ export class GroupGateway {
...
+    const userGroup = await userMember.group.load();
+    let newEvent = (await this.eventService.getEventsByIds([data.eventId]))[0];
+    (await userGroup.members.loadItems()).forEach(
+      async (member: GroupMember) => {
+        //if the user already has the event, keep their tracker
+        let currentUser = await member.user.load();
+        try {
+          let memberEvent =
+            await this.eventService.getCurrentEventTrackerForUser(currentUser);
+          if (memberEvent.event.id != data.eventId)
```
Comment (neketka) [inline server/src/group/group.gateway.ts:90]: I don't think this condition is necessary

File: server/src/group/group.gateway.ts [hunk 3]
```diff
@@ -76,5 +76,32 @@ export class GroupGateway {
...
+      },
+    );
+    userGroup.currentEvent.set(newEvent);
+    await this.groupService.saveGroup(userGroup);
+    let updateGroupData: UpdateGroupDataDto = {
+      curEventId: data.eventId,
+      members: [],
+      removeListedMembers: false,
+    };
+    this.clientService.emitUpdateGroupData(user, updateGroupData);
```
Comment (neketka) [inline server/src/group/group.gateway.ts:105]: Notify all members, not just this one.

**Review by neketka — CHANGES_REQUESTED** [review]
Last few changes, but afterwards this should be ready to go!

**Review by neketka — APPROVED** [review]
LGTM

**Pull request merged**